hexrd/xrdutil/utils.py

```diff
@@ -1101,72 +1101,78 @@ def simulateGVecs(pd, detector_params, grain_params,
-            det_xy[:, 0] <= panel_dims[1][0]
-            )
-        on_panel_y = np.logical_and(
-            det_xy[:, 1] >= panel_dims[0][1],
-            det_xy[:, 1] <= panel_dims[1][1]
-            )
-        on_panel = np.logical_and(on_panel_x, on_panel_y)
-        #
+        det_xy, rMat_s, _ = _project_on_detector_plane(
+            allAngs, rMat_d, rMat_c, chi, tVec_d, tVec_c, tVec_s, distortion
+        )
+
+        on_panel = np.logical_and(
+            np.logical_and(
+                det_xy[:, 0] >= panel_dims[0][0],
+                det_xy[:, 0] <= panel_dims[1][0],
+            ),
+            np.logical_and(
+                det_xy[:, 1] >= panel_dims[0][1],
+                det_xy[:, 1] <= panel_dims[1][1],
+            ),
+        )
+
         op_idx = np.where(on_panel)[0]
-        #
+
         valid_ang = allAngs[op_idx, :]
         valid_ang[:, 2] = xfcapi.mapAngle(valid_ang[:, 2], ome_period)
         valid_ids = allHKLs[op_idx, 0]
         valid_hkl = allHKLs[op_idx, 1:]
         valid_xy = det_xy[op_idx, :]
-        ang_ps = angularPixelSize(valid_xy, pixel_pitch,
-                                  rMat_d, rMat_s,
-                                  tVec_d, tVec_s, tVec_c,
-                                  distortion=distortion)
+        ang_ps = angularPixelSize(
+            valid_xy,
+            pixel_pitch,
+            rMat_d,
+            rMat_s,
+            tVec_d,
+            tVec_s,
+            tVec_c,
+            distortion=distortion,
+        )
 
     return valid_ids, valid_hkl, valid_ang, valid_xy, ang_ps
 
 
-def simulateLauePattern(hkls, bMat,
-                        rmat_d, tvec_d,
-                        panel_dims, panel_buffer=5,
-                        minEnergy=8, maxEnergy=24,
-                        rmat_s=np.eye(3),
-                        grain_params=None,
-                        distortion=None,
-                        beamVec=None):
+@deprecated(new_func=simlp, removal_date='2025-01-01')
+def simulateLauePattern(
+    hkls,
+    bMat,
+    rmat_d,
+    tvec_d,
+    panel_dims,
+    panel_buffer=5,
+    minEnergy=8,
+    maxEnergy=24,
+    rmat_s=np.eye(3),
+    grain_params=None,
+    distortion=None,
+    beamVec=None,
+):
 
     if beamVec is None:
         beamVec = xfcapi.bVec_ref
 
     # parse energy ranges
     multipleEnergyRanges = False
     if hasattr(maxEnergy, '__len__'):
-        assert len(maxEnergy) == len(minEnergy), \
-            'energy cutoff ranges must have the same length'
+        assert len(maxEnergy) == len(
+            minEnergy
+        ), 'energy cutoff ranges must have the same length'
         multipleEnergyRanges = True
-        lmin = []
-        lmax = []
-        for i in range(len(maxEnergy)):
-            lmin.append(processWavelength(maxEnergy[i]))
-            lmax.append(processWavelength(minEnergy[i]))
+        lmin = [processWavelength(e) for e in minEnergy]
+        lmax = [processWavelength(e) for e in maxEnergy]
```

Comment:
It looks like `minEnergy` and `maxEnergy` got swapped in the change here. Can you change them back?

Reply:
Resolved